src/Smalot/PdfParser/Encoding/PDFDocEncoding.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+/**
+ * @file    This file is part of the PdfParser library.
+ *
+ * @author  Brian Huisman <bhuisman@greywyvern.com>
+ *
+ * @date    2023-06-28
+ *
+ * @license LGPLv3
+ *
+ * @url     <https://github.com/smalot/pdfparser>
+ *
+ *  PdfParser is a pdf library written in PHP, extraction oriented.
+ *  Copyright (C) 2017 - Sébastien MALOT <sebastien@malot.fr>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program.
+ *  If not, see <http://www.pdfparser.org/sites/default/LICENSE.txt>.
+ */
+
+// Source : https://opensource.adobe.com/dc-acrobat-sdk-docs/pdfstandards/pdfreference1.2.pdf
+// Source : https://ia801001.us.archive.org/1/items/pdf1.7/pdf_reference_1-7.pdf
+
+namespace Smalot\PdfParser\Encoding;
+
+/**
+ * Class PDFDocEncoding
+ */
+abstract class PDFDocEncoding extends AbstractEncoding
```

Comment:
I dont really like classes which have only 1 function and are also abstract :laughing: Anyway, I suggest we provide access to the long array separately, because it might be helpful to check array content itself sometimes. Also, a method to change its content would be helpful. What do you think?

Reply:
Haha, yeah. I suppose I don't really need to do that. :)

I can add a function that returns the conversion array, but I'm not sure how PdfParser would make use of an additional method to modify it. PDFDocEncoding is a standard encoding documented by Adobe and pretty much set in stone. How would PdfParser benefit from being able to modify it on the fly?

Reply:
> How would PdfParser benefit from being able to modify it on the fly?

The main reason would be flexibility. PDFParser has many issues and there are also many invalid PDFs out there. As a user it might be helpful, if I can tweak around these obstacles by using a custom configuration/setup for a given PDF. I wouldn't go as far as our [CustomConfig](https://github.com/smalot/pdfparser/blob/master/doc/CustomConfig.md) approach here, but it points in the same direction. I am still torn.

Maybe @j0k3r can add something to this as well?

Reply:
Personally speaking, I believe dynamically modifying PDFDocEncoding is not something that is needed, and the main reason is that it only applies to document metadata and other non-document streams. I looked through the entire Issues list, and there are many unresolved encoding problems, but they all have to do with the document stream, and PDFDocEncoding doesn't apply there.

OTOH, adding a function that lets you dynamically modify the conversion table is pretty trivial to implement, so I'll leave it up to your consensus.